### ISSUE TITLE
`import twstock` causes the script.pyw not run.

### ISSUE THREAD

**Issue by seaspeak** [mlouielu/twstock#128]
**Describe the bug**
`import twstock` causes the script.pyw not run.

**To Reproduce**
Or you could use the script below "twstock_not_run.zip"
1. Just incorporate `import twstock` in your script.
2. Rename the script as script.pyw.
3. Double click script.pyw
4. If you comment the `import twstock`, then you could run the script by double-clicking.
5. You could run the script with `twstock` by `python script.pyw`.

**Expected behavior**
Run!

**Screenshots**

**Desktop (please complete the following information):**
 - Windows 10, Python 3.7

**Additional context**

[twstock_not_run.zip](https://github.com/user-attachments/files/16950190/twstock_not_run.zip)

**Comment by mlouielu** [comment]
It works on my Linux.

Please run it in your terminal and see the error message from it.

**Comment by seaspeak** [comment]
script.pyw should be Windows exclusive? Correct me if I am wrong. PYW is run by pythonw.exe so that we could get rid of the black command window.

I found the discussion here, but nothing works for me so far: https://stackoverflow.com/questions/24835155/pyw-and-pythonw-does-not-run-under-windows-7

My other PYW works fine and,
`python script.pyw` works (`python script.py`), but not `pythonw script.pyw`. I don't see any error messages dumped.

**Comment by mlouielu** [comment]
> script.pyw should be Windows exclusive? Correct me if I am wrong. PYW is run by pythonw.exe so that we could get rid of the black command window.
> 

That doesn't matter between platform.

> I found the discussion here, but nothing works for me so far: https://stackoverflow.com/questions/24835155/pyw-and-pythonw-does-not-run-under-windows-7
> 

As I said in the last reply, we should first figure out if there is any error message causing Python to exit early, and I suspect there has some problem of the enviroment or tk.

In order to figure that out, we need to get the terminal first, so that we can get the error message.

> My other PYW works fine and, `python script.pyw` works (`python script.py`), but not `pythonw script.pyw`. I don't see any error messages dumped.
> 

Because pythonw does not show the terminal, thus you can see any error message.

> Simply try twstock_not_run.pyw on Windows, it works without the `import twstock`.

You can try to open up a terminal: https://learn.microsoft.com/en-us/windows/terminal/faq

And then run `python twstock_.pyw`, which should gives you the error message, or if it does not work, rename to `.py` and run it again.

**Comment by seaspeak** [comment]
Let me rephrase the issue, it is Windows specific, pythonw.exe specific. You need to run `pythonw twstock_.pyw` on Windows to reproduce the issue. It was a failure to redirect the stderr. So I rewrite the twstock_.pyw to log the error.
[twstock_.zip](https://github.com/user-attachments/files/16957476/twstock_.zip)

```
  File "C:\Program Files\Python37\lib\site-packages\twstock\cli\best_four_point.py", line 9, in <module>
    getattr(sys.stdout, "buffer", sys.stdout), encoding="utf-8", errors="replace"
AttributeError: 'NoneType' object has no attribute 'readable'
```
Since the pythonw.exe doesn't have the std output, cli module raises the exception. I reassign the `stdout` to `None`. I didn't look into the code, will that impedes your code flow?

```python
# XXX: Repalce sys.stdout prevent Windows UnicodeEncodeError on cmd.exe
try:
	stdout = io.TextIOWrapper(
        getattr(sys.stdout, "buffer", sys.stdout), encoding="utf-8", errors="replace"
	)
except AttributeError:
	stdout = None
```

**Comment by mlouielu** [comment]
> Let me rephrase the issue, it is Windows specific, pythonw.exe specific. You need to run `pythonw twstock_.pyw` on Windows to reproduce the issue. It was a failure to redirect the stderr. So I rewrite the twstock_.pyw to log the error. [twstock_.zip](https://github.com/user-attachments/files/16957476/twstock_.zip)
> 
> ```
>   File "C:\Program Files\Python37\lib\site-packages\twstock\cli\best_four_point.py", line 9, in <module>
>     getattr(sys.stdout, "buffer", sys.stdout), encoding="utf-8", errors="replace"
> AttributeError: 'NoneType' object has no attribute 'readable'
> ```
> 

Thank you, this is what we need to know in order to pin down the problem.

Can you try the branch https://github.com/mlouielu/twstock/tree/fix-pythonw-128 for the fix? You can clone it and copy the "twstock/twstock" directory to the same as your pyw file, or download the zip file, unzip, and copy the "twstock/twstock" directory to the same as your pyw file.

The directory of your pyw file should look like this:

```
foobar/
   |_ your.pyw
   |_ twstock/
```

Then Python should import the twstock with the fix. Please let me know the result, thanks!

**Comment by seaspeak** [comment]
It looks fine.

**Comment by mlouielu** [comment]
My PR didn't pass the unit test, I'll try to fix it later.
